File: cogs/set_active.py
# cogs/cog_set_active.py
import discord
from discord import app_commands
from discord.ext import commands
from utils.data_manager import load_data, save_data

@app_commands.command(name="set_active", description="Define qual torneio será usado como ativo")
async def set_active(self, interaction: discord.Interaction, name: str):
        data = load_data()
        if name not in data["tournaments"]:
            await interaction.response.send_message(" Não existe torneio com esse nome.", ephemeral=True)
            return

        data["active"] = name
        save_data(data)
        await interaction.response.send_message(f" O torneio ativo agora é **{name}**.")

async def setup(bot: commands.Bot):
    await bot.add_cog(SetActiveCog(bot))

File: src/commands/set-active.js
import {
  ApplicationIntegrationType,
  InteractionContextType,
  MessageFlags,
  SlashCommandBuilder,
} from "discord.js";

import { store } from "../data/json-store.js";

export const data = new SlashCommandBuilder()
  .setName("set_active")
  .setDescription("Define qual torneio será usado como ativo")
  .setIntegrationTypes(ApplicationIntegrationType.GuildInstall)
  .setContexts(InteractionContextType.Guild)
  .addStringOption((option) =>
    option
      .setName("name")
      .setDescription("Nome do torneio")
      .setMinLength(1)
      .setMaxLength(100)
      .setRequired(true),
  );

export async function execute(interaction) {
  const name = interaction.options.getString("name", true);
  let missing = false;

  await store.update((state) => {
    if (!Object.hasOwn(state.tournaments, name)) {
      missing = true;
      return;
    }

    state.active = name;
  });

  if (missing) {
    await interaction.reply({
      content: " Não existe torneio com esse nome.",
      flags: MessageFlags.Ephemeral,
    });
    return;
  }

  await interaction.reply(` O torneio ativo agora é **${name}**.`);
}
